src/core/tests/type_prop/reverse_sequence.cpp

```diff
@@ -65,255 +67,274 @@ TEST(type_prop, reverse_sequence_invalid_seq_lengths_et) {
-                  "(rank-static dynamic shape)";
-    } catch (const NodeValidationFailure& error) {
-        EXPECT_HAS_SUBSTRING(
-            error.what(),
-            std::string("Sequence lengths input size (4) is not equal to batch axis dimension of data input (3)"));
-    } catch (...) {
-        FAIL() << "Number of elements of input seq_lengths check with respect batch dimension of input failed for "
-                  "unexpected reason";
-    }
+
+    OV_EXPECT_THROW(
+        auto reverse_seq = std::make_shared<op::v0::ReverseSequence>(data, seq_lengths, batch_axis, seq_axis),
+        NodeValidationFailure,
+        HasSubstr("Sequence lengths input size (4) is not equal to batch axis dimension of data input (3)"));
 }
 
-TEST(type_prop, reverse_sequence_dynamic_invalid_negative_axis_and_data_input_dynamic_rank) {
-    auto data = make_shared<op::Parameter>(element::f32, PartialShape::dynamic());
-    auto seq_lengths = make_shared<op::Parameter>(element::i32, PartialShape{1});
+class TypePropReverseSequenceV0Test : public TypePropOpTest<op::v0::ReverseSequence> {};
+
+TEST_F(TypePropReverseSequenceV0Test, dynamic_invalid_negative_axis_and_data_input_dynamic_rank) {
+    auto data = std::make_shared<Parameter>(element::f32, PartialShape::dynamic());
+    auto seq_lengths = std::make_shared<Parameter>(element::i32, PartialShape{1});
     int64_t batch_axis = 1;
     int64_t seq_axis = -2;
-    try {
-        auto reverse_seq = make_shared<op::ReverseSequence>(data, seq_lengths, batch_axis, seq_axis);
-        FAIL() << "Dynamic rank of data input for negative axis not detected";
-    } catch (const CheckFailure& error) {
-        EXPECT_HAS_SUBSTRING(error.what(), std::string("Rank must be static in order to normalize negative axis=-2"));
-    } catch (...) {
-        FAIL() << "Static rank of data input for negative axis validation check failed for unexpected reason";
-    }
+
+    OV_EXPECT_THROW(auto reverse_seq = make_op(data, seq_lengths, batch_axis, seq_axis),
+                    AssertFailure,
+                    HasSubstr("Rank must be static in order to normalize negative axis=-2"));
+}
+
+TEST_F(TypePropReverseSequenceV0Test, default_ctor) {
+    auto data = std::make_shared<Parameter>(element::f32, PartialShape{4, 3, 2});
+    auto seq_lengths = std::make_shared<Parameter>(element::i32, PartialShape{3});
+    auto op = make_op();
+
+    op->set_arguments(OutputVector{data, seq_lengths});
+    op->set_batch_axis(1);
+    op->set_sequence_axis(0);
+    op->validate_and_infer_types();
+
+    EXPECT_EQ(op->get_batch_axis(), 1);
+    EXPECT_EQ(op->get_sequence_axis(), 0);
+    EXPECT_EQ(op->get_input_size(), 2);
+    EXPECT_EQ(op->get_output_size(), 1);
+    EXPECT_EQ(op->get_output_element_type(0), element::f32);
+    EXPECT_EQ(op->get_output_partial_shape(0), (PartialShape{4, 3, 2}));
+}
+
+TEST_F(TypePropReverseSequenceV0Test, default_ctor_no_arguments) {
+    auto op = make_op();
+    op->set_batch_axis(1);
```

Comment:
`set_sequence_axis` should be tested here as well

Reply:
Is skipped as it has no meaning for shape inference.